src/korma/core.clj

```diff
@@ -548,17 +548,17 @@
    :rpk (raw (eng/prefix child (:pk child)))
    :join-table join-table})
 
-(defn- get-db-keys [parent child]
-  (let [fk-key (default-fk-name parent)]
+(defn- get-db-keys [parent child {:keys [fk]}]
+  (let [fn-key (or fk (default-fk-name parent))]
```

Comment:
fk key provided as an option should be used instead of the default-fk-name call that fails when the belongs-to table is a subselect entity